marshal/consumer.go

```diff
@@ -19,6 +19,28 @@ import (
 	"github.com/dropbox/kafka/proto"
 )
 
+// CommitToken is a minimal structure that contains only the information necessary to
+// mark a message committed. This is done so that you can throw away the message instead
+// of holding on to it in memory.
+type CommitToken struct {
+	topic  string
+	partID int
+	offset int64
+}
+
+// Message is a container for Kafka messages.
+type Message proto.Message
```

Comment:
I'm not sure. Let's try it and if it turns out to be bad we can deal with it, but I think on the consumer side it's probably OK...